### PR TITLE
Transformed --arch switch to --platform for dll tests container scenario

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Transformed --arch switch to --platform for dll tests container scenario

This fix will add support to the `--arch` switch for the scenario where the test dll containers are provided by test file path:

`dotnet test c:\test\...\test.dll --arch x64`

At the moment we support only csproj invocation:

`dotnet test c:\test...\test.csproj --arch x64`

### CODE REVIEW

**Review by nohwnd — APPROVED** [review]

**Review by Haplois — APPROVED** [review]

**Comment by nohwnd** [comment]
This is waiting for approval from servicing.

**Pull request merged**
